R/fixed_design_lf.R

```diff
@@ -0,0 +1,196 @@
+#  Copyright (c) 2023 Merck & Co., Inc., Rahway, NJ, USA and its affiliates.
+#  All rights reserved.
+#
+#  This file is part of the gsDesign2 program.
+#
+#  gsDesign2 is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#' Fixed design sample size
+#'
+#' Computes fixed design sample size for Lachin-Foulkes method.
```

Comment:
As a user, I am confused what does LF method stands for.

A design needs to associate with a specific test. To me LF's paper is a general approach on piecewise enrollment and failure that can be applied to multiple tests.

If the function is related to log rank test with average hazard ration, do we really need this separate function? Or it is already covered by 'fixed_design_ahr'

Reply:
`lf` calls gsDesign functions. It is always hard to tell everything through a function name, so documentation (https://merck.github.io/gsDesign2/articles/usage-fixed-design.html) is important. If the documentation of `fixed_design_xx` is sufficient, then users can always go to the referenced paper for more details.